### PR TITLE
CPP-1823 Get list of files to be hashed from the hooks

### PR DESCRIPTION
**Pull request by apaleslimghost**

### CODE REVIEW

File: core/cli/src/config/hash.ts [hunk 1]
```diff
@@ -17,20 +18,26 @@ export async function fileHash(path: string): Promise<string> {
...
-export async function hasConfigChanged(logger: Logger): Promise<boolean> {
+export async function hasConfigChanged(logger: Logger, config: ValidConfig): Promise<boolean> {
   const hashes = readState('install')
+
   if (!hashes) {
     return true
   }
-  for (const [path, prevHash] of Object.entries(hashes)) {
+
+  for (const path of ['.toolkitrc.yml', ...config.hookManagedFiles]) {
```
Comment (ivomurrell) [inline core/cli/src/config/hash.ts:37]: **suggestion(non-blocking)**: think we could also consider the config to have changed if `Object.keys(hashes) !== ['.toolkitrc.yml', ...config.hookManagedFiles]` to catch cases where a hook has been removed. though idk if that would ever require a reinstall?

File: plugins/circleci/.toolkitrc.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 installs:
-  CircleCi: './lib/circleci-config'
+  CircleCi:
+    entryPoint: './lib/circleci-config'
```
Comment (ivomurrell) [inline plugins/circleci/.toolkitrc.yml:3]: **suggestion(non-blocking)**: should we add the managed files as part of this PR?

```suggestion
    entryPoint: './lib/circleci-config'
    managedFiles:
      - '.circleci/config.yml'
```

File: plugins/package-json-hook/.toolkitrc.yml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 installs:
-  PackageJson: './lib/package-json-helper'
+  PackageJson:
+    entryPoint: './lib/package-json-helper'
```
Comment (ivomurrell) [inline plugins/package-json-hook/.toolkitrc.yml:3]: **suggestion(non-blocking)**: should we add the managed files as part of this PR?

```suggestion
    entryPoint: './lib/package-json-helper'
    managedFiles:
      - 'package.json'
```

**Review by ivomurrell — APPROVED** [review]
nice!

File: plugins/circleci/.toolkitrc.yml (hunk 2 above)
Comment (apaleslimghost) [reply plugins/circleci/.toolkitrc.yml:3]: oops, meant to do that but forgot!

File: core/cli/src/config/hash.ts (hunk 1 above)
Comment (apaleslimghost) [reply core/cli/src/config/hash.ts:37]: it's impossible for a _removed_ hook to cause a difference with install so i don't think we need this

**Pull request merged**